test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -0,0 +1,284 @@
+			.getByRole( 'tab', { name: 'Styles' } )
+			.click();
+
+		// Ensure there the default value for the minimum height of cover is undefined.
+		const defaultHeightValue = await coverBlockEditorSettings
+			.getByLabel( 'Minimum height of cover' )
+			.inputValue();
+		expect( defaultHeightValue ).toBeFalsy();
+
+		// There is no accessible locator for the draggable block resize edge,
+		// which is he bottom edge of the Cover block.
+		// Therefore a CSS selector must be used.
+		const coverBlockResizeHandle = page.locator(
+			'.components-resizable-box__handle-bottom'
+		);
+
+		// Establish the existing bounding boxes for the Cover block
+		// and the Cover block's resizing handle.
+		const coverBlockBox = await coverBlock.boundingBox();
+		const coverBlockResizeHandleBox =
+			await coverBlockResizeHandle.boundingBox();
+		expect( coverBlockBox.height ).toBeTruthy();
+		expect( coverBlockResizeHandleBox.height ).toBeTruthy();
+
+		// Increse the Cover block height by 100px.
+		await coverBlockResizeHandle.hover();
+		await page.mouse.down();
+
+		// Counter-intuitively, the mouse movement calculation should not be made using the
+		// Cover block's bounding box, but rather based on the coordinates of the
+		// resize handle.
+		await page.mouse.move(
+			coverBlockResizeHandleBox.x + coverBlockResizeHandleBox.width / 2,
+			coverBlockResizeHandleBox.y + 100
+		);
+		await page.mouse.up();
+
+		const newCoverBlockBox = await coverBlock.boundingBox();
+		expect( newCoverBlockBox.height ).toBe( coverBlockBox.height + 100 );
+	} );
+
+	test( 'dims the background image down by 50% when transformed from the Image block', async ( {
+		page,
+		editor,
+		imageBlockUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/image' } );
+
+		const imageBlock = page.getByRole( 'document', {
+			name: 'Block: Image',
+		} );
+
+		await imageBlockUtils.upload(
+			imageBlock.getByTestId( 'form-file-upload-input' )
+		);
+
+		await expect(
+			page
+				.getByRole( 'document', { name: 'Block: Image' } )
+				.locator( 'img' )
+		).toBeVisible();
+
+		await editor.transformBlockTo( 'core/cover' );
+
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		// The hidden span must be used as the target for opacity and color value.
+		// Using the Cover block to calculate the opacity results in an incorrect value of 1.
+		// The hidden span value returns the correct opacity at 0.5.
+		const [ backgroundDimColor, backgroundDimOpacity ] =
+			await imageBlockUtils.getBackgroundColorAndOpacity(
+				coverBlock.locator( 'span[aria-hidden="true"]' )
+			);
+
+		// The hidden span must be used as the target for opacity and color value.
+		// Using the Cover block to calculate the opacity results in an incorrect value of 1.
+		// The hidden span value returns the correct opacity at 0.5.
+		expect( backgroundDimColor ).toBe( 'rgb(0, 0, 0)' );
+		expect( backgroundDimOpacity ).toBe( '0.5' );
+	} );
+} );
+
+class ImageBlockUtils {
```

Comment:
Maybe we should name it `CoverBlockUtils`? 😆

Reply:
Makes sense. I thought maybe `ImageBlockUtils` was always used as the name for the POM class, for some reason.

Reply:
Name changed to `CoverBlockUtils` in 405c70909d0fd3944db7a518805e48256eeb8fd6.